clients/js-legacy/package.json

```
@@ -10,16 +10,31 @@
         "node": ">=16"
     },
     "files": [
-        "src"
+        "lib",
+        "src",
+        "LICENSE",
+        "README.md"
     ],
+    "publishConfig": {
+        "access": "public"
+    },
+    "main": "./lib/cjs/index.js",
+    "module": "./lib/esm/index.js",
+    "types": "./lib/types/index.d.ts",
+    "exports": {
+        "types": "./lib/types/index.d.ts",
+        "require": "./lib/cjs/index.js",
+        "import": "./lib/esm/index.js"
+    },
     "scripts": {
         "prepublishOnly": "pnpm build",
-        "build": "tsc --build --verbose",
+        "build": "tsc --build --verbose tsconfig.all.json",
         "lint": "eslint --max-warnings 0 .",
         "lint:fix": "eslint --fix .",
         "format": "prettier --check src test",
         "format:fix": "prettier --write src test",
-        "test": "mocha test/*"
+        "test": "mocha test/*",
+        "test:exports": "node ./test/exports/module.mjs && node ./test/exports/commonjs.cjs"
```

You may want to add the `test:exports` script inside the `test` script. Something like that:

```suggestion
        "test": "mocha test/* && pnpm test:exports",
        "test:exports": "node ./test/exports/module.mjs && node ./test/exports/commonjs.cjs"
```

---

Oh right, good idea!